frontend/pages/[resourceType].tsx

```diff
@@ -0,0 +1,86 @@
+import { GetServerSideProps } from 'next';
+import { Button, Grid, Divider } from '@mantine/core';
+import Link from 'next/link';
+
+/**
+ * Component which displays list of all resources of some type as passed in by (serverside) props
+ * @returns component with list of resource (by id) buttons that are links to that resource's details
+ */
+export default function ResourceList({ ids, resourceType }: { ids: Array<string>; resourceType: string }) {
+  const idItems = ids.map(id => {
+    return (
+      <Link href={`/${resourceType}/${id}`} key={id}>
+        <div>
+          <Button
+            color="cyan"
+            radius="md"
+            size="md"
+            variant="subtle"
+            style={{
+              padding: '2px'
+            }}
+          >
+            <div>
+              {resourceType}/{id}
+            </div>
+          </Button>
+        </div>
+      </Link>
+    );
+  });
+  return (
+    <div
+      style={{
+        width: '78vw'
+      }}
+    >
+      <Grid columns={7}>
+        <Grid.Col offset={3} span={2} style={{ paddingTop: '5px' }}>
+          <h2 style={{ color: 'gray', marginTop: '0px', marginBottom: '8px' }}>{`${resourceType} IDs`}</h2>
+        </Grid.Col>
+        <Grid.Col
+          span={2}
+          style={{
+            paddingTop: '5px'
+          }}
+        ></Grid.Col>
+      </Grid>
+      <Divider my="md" style={{ marginTop: '14px' }} />
+      <div>
+        <div
+          style={{
+            textAlign: 'left',
+            overflowWrap: 'break-word',
+            height: '500px',
+            padding: '10px',
+            backgroundColor: '#FFFFFF',
+            border: '1px solid',
+            borderColor: '#DEE2E6',
+            borderRadius: '20px',
+            marginTop: '10px',
+            marginBottom: '20px',
+            marginLeft: '150px',
+            marginRight: '150px'
+          }}
+        >
+          <ul>{idItems}</ul>
+        </div>
+      </div>
+    </div>
+  );
+}
+
+/**
+ * Serverside props pulls id data of a certain resourceType to pass to the page before it's sent to browser
+ * @returns props for the [resourceType] page that pass resourceType and ids of resources of that type
+ */
+export const getServerSideProps: GetServerSideProps = async context => {
```

Comment:
Do we want to have a loading spinner of some kind while we are fetching the resource data?

Reply:
Loading spinners only make sense during client side fetching. With `getServerSideProps`, nothing is rendered on the page until the resource data is returned. This function runs on the server before returning the HTML that gets sent back to the client, so there will be no loading state